lib/plug/conn.ex

```diff
@@ -446,13 +446,13 @@ defmodule Plug.Conn do
   `:sent`.
   """
   @spec put_resp_header(t, binary, binary) :: t
-  def put_resp_header(%Conn{resp_headers: headers, state: state} = conn, key, value) when
-      is_binary(key) and is_binary(value) and state != :sent do
-    %{conn | resp_headers: List.keystore(headers, key, 0, {key, value})}
+  def put_resp_header(%Conn{state: :sent}, _key, _value) do
```

Comment:
Sorry, we should just check for `:sent` then. `put_status` is also wrong, it should only check for `:sent` too.

The previous places that used `@unsent` though are correct (you should not call `resp` or `send_resp` when `:chunked` or `:file` is set). So we should not change `@unsent`.